src/pages/dataElements/form/customFields.tsx

```diff
@@ -0,0 +1,357 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    ButtonStrip,
+    Button,
+    Field,
+    Radio,
+    SingleSelectFieldFF,
+} from '@dhis2/ui'
+import React, { useRef } from 'react'
+import { Field as FieldRFF, useField } from 'react-final-form'
+import { useHref } from 'react-router'
+import {
+    AggregationLevelMultiSelect,
+    ColorAndIconPicker,
+    CategoryComboSelect,
+    OptionSetSelect,
+    LegendSetTransfer,
+} from '../../../components'
+import {
+    useSchemas,
+    translateAggregationType,
+    translateValueType,
+} from '../../../lib'
+import classes from './customFields.module.css'
+import { EditableFieldWrapper } from './EditableFieldWrapper'
+
+// @TODO(custom fields): Implement me! (1)
+export function ColorAndIconField() {
```

Comment:
I guess we will move this out of `/dataElements/form` and into a generic component, that just takes a `name`-prop that it would pass to `useField`?

Reply:
I guess that depends, although it's quite likely.
In this regard I favor [waiting for emerging patterns](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) ("_Always implement things when you actually need them, never when you just foresee that you [will] need them._") and then refactoring the patterns into shared concretions.
At some point we'll probably have a folder with fields [like in the import export app](https://github.com/dhis2/import-export-app/tree/master/src/components/Inputs).

Reply:
I mean, we will have these for a lot of objects though? We know what's going to be implemented down the line. There's a difference between waiting when you don't know, and actually know because we're recreating something.

But again, this is fine for now, I can go over and generalize things.